core/commands/add.go

```diff
@@ -317,6 +317,12 @@ You can now check what blocks have been created by:
 				return nil
 			}
 
+			if !local {
+				if err := n.Providers.ProvideRecursive(req.Context(), nd, dserv); err != nil {
```

Comment:
@whyrusleeping I think it's the last thing I wanted to discuss after removing the [WIP] tag:

Before strategies are implemented (next PR), this will happen on the end of add causing weird hang for bigger adds. Should this be done in background for now (`go Provide..`)?

Reply:
I think its probably fine for now. But in the future I think we will want a separate progress bar for providing.